src/main/src/main/java/org/geoserver/ows/StylePublisher.java

```diff
@@ -75,7 +85,10 @@ protected URL getUrl(HttpServletRequest request) throws IOException {
 
             switch (resource.getType()) {
                 case RESOURCE:
-                    return URLs.fileToUrl(resource.file());
+                    // don't allow access to the style .xml files
+                    return resource.name().endsWith(".xml")
+                            ? null
+                            : URLs.fileToUrl(resource.file());
```

Comment:
There is no easy way to tell if an xml document in the styles directory is part of the catalogue configuration, or just an xml document. So checking for "xml" extension is a fair assumption.

Technically an SLD file is just an xml document, and it is would be okay to upload such a file as a style for use.

Reply:
There really shouldn't be any .xml files in the style directories that isn't a valid StyleInfo file since every one is going to be an XStream stack trace in the logs on every GeoServer restart/reload.

Reply:
Wow okay :)